backend/src/services/model.ts

```diff
@@ -347,9 +346,27 @@ export async function createModelCardFromSchema(
 }
 
 export async function createModelCardFromTemplate(
-  _user: UserInterface,
-  _modelId: string,
-  _schemaId: string,
+  user: UserInterface,
+  modelId: string,
+  templateId: string,
 ): Promise<ModelCardRevisionDoc> {
-  throw NotImplemented({}, 'This feature is not yet implemented')
+  if (modelId === templateId) {
+    throw BadReq('The model and template ID must be different', { modelId, templateId })
+  }
+  const model = await getModelById(user, modelId)
+  checkModelRestriction(model)
```

Comment:
Do we need to `checkModelRestriction` for the model being used as a template also?

Reply:
I don't think so. All it does is check to see if it's "mirrored" which is basically read only. As we're not editing the template in any way, I think it's fine to use it if it's read only.